game-core/src/main/java/games/strategy/triplea/odds/calculator/OddsCalculatorPanel.java

```diff
@@ -595,24 +597,18 @@ private void updateStats() {
         if (isLand()) {
           bombarding = CollectionUtils.getMatches(attacking, Matches.unitCanBombard(getAttacker()));
           attacking.removeAll(bombarding);
+          final int numLandUnits = CollectionUtils.countMatches(attacking, Matches.unitIsLand());
+          if (Properties.getShoreBombardPerGroundUnitRestricted(data) && numLandUnits < bombarding.size()) {
+            BattleDelegate.sortUnitsToBombard(bombarding);
+            bombarding = new ArrayList<>(bombarding.subList(0, numLandUnits));
```

Comment:
Is bombarding modified afterwards?
If this is the case it would make sense to document (with a comment) why a copy is created.

Reply:
subList returns a non-serializable list which then causes issues with game data.

Reply:
Ah, makes sense. Would you mind adding a comment explaining that?

Reply:
Done.